From ec34b80ca6ebd482968b1216553456aefabacd8e Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 14 Mar 2021 16:46:52 +0200
Subject: [PATCH] [SW-34097] i2c_slave: prevent disabling i2c in test mode

when the device is in hw test mode, i2c slave interface must not be
disabled

Change-Id: Ia4d1e549dea6feda3006879c9c2f0530efc59bd7
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 3c9bca4..6f01dc4 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -199,6 +199,11 @@
 void module_i2c_s_disable(void)
 {
 	u32 val;
+	bool debug_mode;
+
+	debug_mode = hl_is_debug_mode(HL_DEBUG_HW_I2C);
+	if (debug_mode)
+		return;
 
 	i2cslave_flush();
 	regmap_writel(0x0, mmPSOC_I2C_S_IC_ENABLE);
@@ -421,6 +426,11 @@
 void module_i2c_s_teardown(void)
 {
 	u32 val;
+	bool debug_mode;
+
+	debug_mode = hl_is_debug_mode(HL_DEBUG_HW_I2C);
+	if (debug_mode)
+		return;
 
 	/* Write 0 to IC_ENABLE to disable DW_apb_i2c */
 	regmap_writel(0, mmPSOC_I2C_S_IC_ENABLE);
